[sprint_7-solution] add backup file with tasks

PascalABCD opened the pull request:
- add FileBackedTaskManager to save tasks and load tasks from file
- add tests for saved and loaded tasks

PascalABCD committed:
[sprint_7-solution] add backup file with tasks

- add FileBackedTaskManager to save tasks and load tasks from file
- add tests for saved and loaded tasks

irek-valiev commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,250 @@
+        this.file = file;
+    }
+
+    public void save() {
```
метод не должен быть публичным, так как является частью реализации и не должен быть виден или доступен для внешних потребителей

irek-valiev commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,250 @@
+                } else {
+                    manager.tasks.put(task.getId(), task);
+                }
+            }
```
Сейчас вычитанные из файла Эпики потеряли данные о привязанных к ним подзадачах. Чтобы наполнить внутреннее хранилище Эпиков (subtasksList) нужно пройтись по всем подзадачам из хранилища subtasks, по значению epicId брать конкретный Эпик из хранилища epics и использовать метод addSubtask()

irek-valiev commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,250 @@
+            System.out.println(e.getMessage());
+        }
+    }
+
```
не нужно переопределять родительские методы, если в них ничего не меняется

PascalABCD committed:
[sprint_7-solution] review fixes

- while load tasks, add subtasks to epics
- fix tests to cover new logic
- remove override from getAll* methods
- make save() private

PascalABCD committed:
[sprint_7-solution] review fixes

- while load tasks, add subtasks to epics
- fix tests to cover new logic
- remove override from getAll* methods
- make save() private

irek-valiev commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -196,7 +205,6 @@ public void removeAllSubtasks() {
     }
 
-    @Override
     public List<Subtask> getAllSubtasks() {
```
методы getAllTasks(), getAllSubtasks() и getAllEpics() в FileBackedTaskManager  нужно удалить. Все методы родительского класса доступны в дочерних и переопределяются те, в которых нужно что-то дополнить

PascalABCD committed:
[sprint_7-solution] remove getAll* methods from FileBackedTaskManager
